sycl-jit/jit-compiler/lib/rtc/DeviceCompilation.cpp

```diff
@@ -831,30 +831,38 @@ Error jit_compiler::linkDeviceLibraries(llvm::Module &Module,
   DiagnosticsEngine Diags(DiagID, DiagOpts, Wrapper.consumer(),
                           /* ShouldOwnClient=*/false);
 
+  Triple T{Module.getTargetTriple()};
+  Driver D{(SYCLToolchain::instance().getPrefix() + "/bin/clang++").str(),
+           T.getTriple(), Diags};
+
   SmallVector<std::string> LibNames;
   getDeviceLibraries(UserArgList, LibNames, Format);
   const bool IsCudaHIP =
       Format == BinaryFormat::PTX || Format == BinaryFormat::AMDGCN;
   if (IsCudaHIP) {
     // Based on the OS and the format decide on the version of libspirv.
     // NOTE: this will be problematic if cross-compiling between OSes.
-    std::string Libclc{"clc/"};
-    Libclc.append(
 #ifdef _WIN32
-        "remangled-l32-signed_char.libspirv-"
+    std::string Libclc = "remangled-l32-signed_char.libspirv-";
 #else
-        "remangled-l64-signed_char.libspirv-"
+    std::string Libclc = "remangled-l64-signed_char.libspirv-";
 #endif
-    );
     Libclc.append(Format == BinaryFormat::PTX ? "nvptx64-nvidia-cuda.bc"
                                               : "amdgcn-amd-amdhsa.bc");
     LibNames.push_back(Libclc);
   }
 
   LLVMContext &Context = Module.getContext();
   for (const std::string &LibName : LibNames) {
-    std::string LibPath =
-        (SYCLToolchain::instance().getPrefix() + "/lib/" + LibName).str();
+    std::string LibPath;
+    if (LibName.find("libspirv") != std::string::npos) {
+      SmallString<256> LibraryPath(D.ResourceDir);
```

Comment:
IMO, resource dir getter should go to the `SYCLToolchain` class, maybe it should be even pre-computed and stored as a data member.